synapse/models/infotech.py

```diff
@@ -303,6 +450,46 @@ def getBaseModels():
                     ('reg:int', {'ptype': 'it:dev:int', 'doc': 'The integer value that was deleted (parsed from :reg).', 'ro': 1}),
                     ('reg:bytes', {'ptype': 'file:bytes', 'doc': 'The binary data that was deleted (parsed from :reg).', 'ro': 1}),
                 )),
+                ('it:prod:soft', {}, (
+                    ('name', {'ptype': 'str:lwr', 'ro': 1, 'req': 1,
+                              'doc': 'Name of the software'}),
+                    ('desc', {'ptype': 'str:txt', 'doc': 'A description of the software'}),
+                    ('desc:short', {'ptype': 'str:lwr', 'doc': 'A short description of the software'}),
+                    ('author:org', {'ptype': 'ou:org', 'doc': 'Organization responsible for the software', }),
+                    ('author:acct', {'ptype': 'inet:web:acct', 'doc': 'Web user responsible for the software', }),
+                    ('author:person', {'ptype': 'ps:person', 'doc': 'Person responsible for the software', }),
+                    ('url', {'ptype': 'inet:url', 'doc': 'URL relevant for the software', }),
+                )),
+
+                ('it:prod:softver', {}, (
+                    ('software', {'ptype': 'it:prod:soft', 'req': 1, 'ro': 1,
```

Comment:
it:prod:softver currently doesn't have a meaningful way to repr due to not knowing the name of the software family. Perhaps we should encode the current :name from it:prod:soft?  Seems like there are issues either way ( about duplicate data etc ).  Maybe the answer is to pull :name off the adjacent it:prod:soft at the time softver is created and create it:prod:softver:name to show that was it's name at the time of that version?

Reply:
that's pretty easy to grab in a callback function  if its not set (and we have other cases of doing that already iirc).

Reply:
actually encoding the current :name is pretty tight since it allows us to track name changes over time for a given product.

Reply:
fixed